specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json

```diff
@@ -117,6 +117,127 @@
+            "x-ms-skip-url-encoding": true
+          },  
+          {
+            "name": "grain",
+            "description": "Can be daily or monthly",
+            "in": "query",
+            "required": true,
+            "type": "string"
+          },       
+          {
+            "name": "$filter",
+            "description": "Required only for daily grain. The properties/UsageDate for start date and end date. The filter supports 'le' and  'ge'",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationSummariesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }, 
+    "/{scope}/providers/Microsoft.Consumption/reservationdetails": {
+      "get": {
+        "tags": [
+          "ReservedInstances"
+        ],
+        "operationId": "ReservationsDetails_List",
+        "description": "Lists the reservations details for provided date range.",
+        "externalDocs": {
+          "url": "https://azure.microsoft.com/en-us/blog/announcing-preview-of-consumption-and-charge-apis-for-enterprise-azure-customers/"
+        },
+        "x-ms-examples": {         
+          "ReservationDetails": {
+            "$ref": "./examples/ReservationDetails.json"
+          },
+          "ReservationDetailsWithReservationId": {
+            "$ref": "./examples/ReservationDetailsWithReservationId.json"
+          }
+          },
+        "parameters": [
+          {
+            "name": "scope",
+            "description": "The scope of the reservation details. The scope can be 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}' or 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}/reservations/{ReservationId}'",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true
+          },              
+          {
+            "name": "$filter",
+            "description": "Filter reservation details by date range. The properties/UsageDate for start date and end date. The filter supports 'le' and  'ge' ",
+            "in": "query",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationDetailsListResult"
```

Comment:
same comment as previous, would you want to mark this as x-ms-pageable?

Reply:
pushed my changes